app/assets/javascripts/app_router.js

```diff
@@ -1,22 +1,48 @@
 var Pokedex = require("./models/pokedex.js");
 var PokedexView = require("./pokedex_view.js");
+var PokemonShowView = require("./pokemon_show_view.js");
 
-var AppRouter = function ($element) {
+var AppRouter = function (routes) {
   this.pokedex = new Pokedex();
-  this.pokedex.fetchThemAll(this.renderPokedex.bind(this));
-	this.$el = $element;
-  this.renderPokedex();
+  this.routes = routes;
+	this.container;
```

Comment:
when we create the router we pass it an object with some routes { "pokedex": "renderPokedexView" ... }

So that when we use the method router.go("pokedex") we check on the this.routes object to see which method we have to call for that route.

Does that make sense?

And yes, sorry we should keep proper indentation!